src/Core/Framework/Adapter/Translation/Translator.php

```diff
@@ -83,6 +84,10 @@ public function __construct(
 
     public static function buildName(string $id): string
     {
+        if (\strpbrk($id, ItemInterface::RESERVED_CHARACTERS) !== false) {
```

Comment:
can you please add a testcase for this? 

And also i'm wondering if there may be an easier solution, e.g. just using an md5 of the id here, as we have a lot of places where we build cache item keys and currently none of them has this special checks, but probably all the other places only work with hex ids or hashes, so it should not be a problem

Reply:
@keulinho I added a really simple unit test on the buildName function. Do you want a more complex test integrating the Cache, too?

Regarding your ideas:

1. Would you hash all the IDs? Or just in case it contains reserved characters? The translation cache tags are indeed quite weird and long as they just contain language snippets. But hashing those makes debugging harder.
2. The buildName is quite similar for a lot of Cached elements - this fix only applies for translations. Actually other cache tags can also contain reserved characters (theoretically)
![2022-10-26 at 09 04](https://user-images.githubusercontent.com/10775702/197957467-b6a59a5a-e047-4c1f-ad54-175bdf9ee951.png)